src/PollingBlockTracker.ts

```diff
@@ -106,9 +108,17 @@ export class PollingBlockTracker
       return this._currentBlock;
     }
     // wait for a new latest block
-    const latestBlock: string = await new Promise((resolve) =>
-      this.once('latest', resolve),
-    );
+    const latestBlock: string = await new Promise((resolve) => {
```

Comment:
To my understanding this change allows the `PollingBlockTracker` instance to be shut down (i.e. network calls will stop). 

Though this promise here will never reject, meaning that in the case of:
- `PollingBlockTracker` being stopped, and
- Desired network not available

The promise will be hanging forever.

I believe the fix should also reject this promise in the above scenario, resuming the caller execution

Reply:
Done in 3f39061b910606419ebe0b969fabfebe2c3a6b52

Reply:
Interesting, good catch!

What do you think of this alternative solution? https://github.com/MetaMask/eth-block-tracker/pull/286/
It seemed like avoiding additional internal listeners might be simpler here. And I am also a bit hesitant to leave listeners intact in `destroy`, because that might result in a memory leak.